src/main/java/Duke.java

```diff
@@ -1,10 +1,297 @@
+            else if(startsWith(input, "deadline")){
+                runDeadline(taskList, parseJob(input, "/by"), parseDate(input, "/by"));
+            }
+
+            // EVENTS COMMAND
+            else if(startsWith(input, "event")){
+                runEvent(taskList, parseJob(input, "/at"), parseDate(input, "/at"));
+            }
+
+            // ADD command
+            else {
+                runAdd(taskList, input);
+            }
+        }
+
+
+    }
+
+    /**
+     * COMMAND RUNNER METHODS
+     * */
+
+    private static void runAdd(Task[] taskList, String input) {
+        // stores user command as job
+        Task newTask = new Task(input);
+
+        taskList[Task.taskCount] = newTask;
+        Task.taskCount++;
+
+        printTaskAdded(newTask);
+    }
+
+    private static void runDone(Task[] taskList, String input) {
+        String[] word = input.split(" ");
+
+        // check invalid input
+        if(word.length != 2){
+            printInvalidInputWarning();
+            return;
+        }
+
+        int jobNumber = Integer.parseInt(word[1]) - 1;
+
+        if(jobNumber < Task.taskCount && jobNumber >= 0){
+            markJobAsDone(taskList[jobNumber]);
+        }
+        else if(Task.taskCount == 0){
+            printNoTaskWarning();
+        }
+        else{
+            printInvalidTaskWarning(jobNumber);
+        }
+    }
+
+    private static void runList(Task[] taskList) {
+        int numbering = 1;
+
+        // error handling - no jobs
+        if(Task.taskCount == 0){
+            System.out.println("No tasks yet! What would you like to do today?\n");
+        }
+        else {
+            for (int i = 0; i < Task.taskCount; i++) {
+                System.out.print(numbering + ". ");
+                taskList[i].printTask();
+                numbering++;
+            }
+            System.out.println();
+        }
+    }
+
+    private static void runTodo(Task[] taskList, String input) {
+
+        if(input == null){
+            printInvalidInputWarning();
+            return;
+        }
+
+        Todo newTask = new Todo(input);
+
+        taskList[Task.taskCount] = newTask;
+        Task.taskCount++;
+
+        printTaskAdded(newTask);
+    }
+
+    private static void runDeadline(Task[] taskList, String input, String by){
+
+        if(input == null || by == null){
+            printInvalidInputWarning();
+            return;
+        }
+
+        Deadline newTask = new Deadline(input, by);
+        taskList[Task.taskCount] = newTask;
+        Task.taskCount++;
+
+        printTaskAdded(newTask);
+
+    }
+
+    private static void runEvent(Task[] taskList, String input, String by){
+
+        if(input == null || by == null){
+            printInvalidInputWarning();
+            return;
+        }
+
+        Event newTask = new Event(input, by);
+        taskList[Task.taskCount] = newTask;
+        Task.taskCount++;
+
+        printTaskAdded(newTask);
+
+    }
+
+    /**
+     * IDK TO CLASSIFY
```

Comment:
Personal suggestion - Maybe these functions can be classified under "read and parse user input"?

Reply:
Nice, thank you for the suggestion!